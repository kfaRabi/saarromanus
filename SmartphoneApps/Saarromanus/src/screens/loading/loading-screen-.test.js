import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import LoadingScreen from './index';

const mockStore = configureStore();

const mockFn = jest.fn();

const initialLoadingState = {
	config: { loading: true },
};

const initialNonLoadingState = {
	config: { loading: false },
};

describe('<LoadingScreen />', () => {
	describe('If Not Loading', () => {
		let wrapper;
		let store;

		beforeAll(() => {
			store = mockStore(initialNonLoadingState);
			wrapper = shallow(<LoadingScreen store={store} />).dive();
		});

		it('should not render', () => {
			expect(wrapper).toMatchSnapshot('loading:false');
			expect(wrapper.dive().children().length).toBeFalsy();
		});

		it('should not mount any <View /> component', () => {
			expect(wrapper.find('View').length).toBe(0);
		});
	});
	describe('When Loading', () => {
		let wrapper;
		let store;
		let component;

		beforeAll(() => {
			store = mockStore(initialLoadingState);
			wrapper = shallow(<LoadingScreen store={store} />).dive();
			component = wrapper.dive();
		});

		it('should render properly', () => {
			expect(wrapper).toMatchSnapshot('loading:true');
		});

		it('should mount <View /> component', () => {
			expect(component.find('View').length).toBe(1);
		});

		it('should render two <Image /> components', () => {
			expect(component.find('Image').length).toBe(2);
		});

		it('should render a <ActivityIndicator /> component', () => {
			expect(component.find('ActivityIndicator').length).toBe(1);
		});

		it('should have three inner components', () => {
			expect(component.find('View').children().length).toBe(3);
		});
	});
});
