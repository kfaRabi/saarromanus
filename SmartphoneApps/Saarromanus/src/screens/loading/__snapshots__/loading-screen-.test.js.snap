// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingScreen /> If Not Loading should not render: loading:false 1`] = `
<LoadingScreen
  loading={false}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;

exports[`<LoadingScreen /> When Loading should render properly: loading:true 1`] = `
<LoadingScreen
  loading={true}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
